Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -126,11 +128,17 @@ public function commit(string $streamName, array $data, int $commitVersion, \Clo
 
         $version = 1;
         array_map(function (EventTransport $eventTransport) use ($query, &$version) {
-            $event = $this->serializer->serialize($eventTransport->getEvent());
-            $metadata = $this->serializer->serialize($eventTransport->getMetadata());
+            $convertedEvent = $this->propertyMapper->convert($eventTransport->getEvent(), 'array');
+
+            // HACK this should go into a separate column in the format "<BoundedContext>:<EventType>"
+            $convertedEvent['__type'] = TypeHandling::getTypeForValue($eventTransport->getEvent());
```

Comment:
I just manually add the `__type` here so we can unserialize it in `unserializeEvents()` but we should solve this otherwise IMO